File: {{ cookiecutter.repo_name }}/{{ cookiecutter.module_name }}/data/dataset.py
from pathlib import Path

from loguru import logger
from tqdm import tqdm

from {{ cookiecutter.module_name }}.config import PROCESSED_DATA_DIR, RAW_DATA_DIR


def main(
    input_path: Path = RAW_DATA_DIR / "dataset.csv",
    output_path: Path = PROCESSED_DATA_DIR / "dataset.csv",
) -> None:
    """Process dataset from raw to processed format.

    Parameters
    ----------
    input_path : Path
        Path to input dataset.
    output_path : Path
        Path to save processed dataset.
    """
    logger.info("Processing dataset...")
    for i in tqdm(range(10), total=10):
        if i == 5:
            logger.info("Something happened for iteration 5.")
    logger.success("Processing dataset complete.")


if __name__ == "__main__":
    main()

File: {{ cookiecutter.repo_name }}/src/data/dataset.jl
"""
Process dataset from raw to processed format.
Mirrors {{ cookiecutter.module_name }}/data/dataset.py
"""

using DataFrames, CSV
using Logging

include("../config.jl")

# === Configuration ===
const INPUT_FILE = joinpath(RAW_DATA_DIR, "dataset.csv")
const OUTPUT_FILE = joinpath(PROCESSED_DATA_DIR, "dataset.arrow")

# === Load Data ===
@info "Loading raw data from $INPUT_FILE"

# Read from original format
# df = CSV.read(INPUT_FILE, DataFrame)

# === Processing ===
@info "Processing dataset..."

for i in 1:10
    if i == 5
        @info "Something happened for iteration 5."
    end
end

# === Save Results ===
@info "Saving processed data to $OUTPUT_FILE"
# Arrow.write(OUTPUT_FILE, df)

@info "Processing complete. Saved to Arrow format."
